process-kafka-quickstart-quarkus/src/test/java/org.acme.travel/KafkaTester.java

```diff
@@ -0,0 +1,101 @@
+import org.apache.kafka.clients.producer.ProducerConfig;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.common.serialization.IntegerDeserializer;
+import org.apache.kafka.common.serialization.IntegerSerializer;
+import org.apache.kafka.common.serialization.StringDeserializer;
+import org.apache.kafka.common.serialization.StringSerializer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class KafkaTester {
+
+    private KafkaProducer<String, String> producer;
+    private KafkaConsumer<String, String> consumer;
+    private CountDownLatch shutdownLatch = new CountDownLatch(1);
+    private AtomicBoolean shutdown = new AtomicBoolean(false);
+    private static Logger LOGGER = LoggerFactory.getLogger(KafkaTester.class);
+
+    public KafkaTester(String hosts) {
+        Properties producerConfig = new Properties();
+        producerConfig.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, hosts);
+        producerConfig.put(ProducerConfig.CLIENT_ID_CONFIG, this.getClass().getName() + "Producer");
+        producerConfig.put(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, IntegerSerializer.class.getName());
+        producerConfig.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, StringSerializer.class.getName());
+        producer = new KafkaProducer<>(producerConfig);
+
+        Properties consumerConfig = new Properties();
+        consumerConfig.put(ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG, "true");
+        consumerConfig.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest");
+        consumerConfig.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, hosts);
+        consumerConfig.put(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, IntegerDeserializer.class.getName());
+        consumerConfig.put(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, StringDeserializer.class.getName());
+        consumerConfig.put(ConsumerConfig.GROUP_ID_CONFIG, this.getClass().getName() + "Consumer");
+        consumer = new KafkaConsumer<>(consumerConfig);
+    }
+
+    public void consume(String topic, Consumer<String> callback) {
+        consumer.subscribe(Collections.singletonList(topic));
+
+        CompletableFuture.runAsync(() -> {
+            while (!shutdown.get()) {
+                ConsumerRecords<String, String> records = consumer.poll(Duration.ofMillis(500));
+
+                StreamSupport.stream(records.spliterator(), true)
+                        .map(ConsumerRecord::value)
+                        .forEach(callback::accept);
+
+                consumer.commitSync();
+            }
+            shutdownLatch.countDown();
+        });
+    }
+
+    public void produce(String data, String topic) {
+        producer.send(new ProducerRecord<>(topic, data), (m, ex) -> {
+            Optional.ofNullable(ex).ifPresent(e -> LOGGER.error("Error publishing message {}", m, ex));
+        });
+    }
+
+    public void shutdown() {
+        try {
+            shutdown.set(true);
+            shutdownLatch.await();
```

Comment:
@tiagodolphine maybe use a timeout to not wait forever
```suggestion
            shutdownLatch.await(1, TimeUnit.MINUTES);
```
and maybe move consumer and produce to a separated block as the latch can still throw an exception.

Reply:
+1

Reply:
done